### PR TITLE
feat(storage): functional storage library options

### PR DESCRIPTION
**Pull request by AllanZhengYP**
#### Description of changes

Add Storage-specific configuration interfaces and library options interfaces

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/singleton/Storage/types.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+export type StorageAccessLevel = 'guest' | 'protected' | 'private';
+
+export interface StorageConfig {
+	bucket?: string;
+	region?: string;
+}
+
+export type StoragePrefixResolver = (params: {
```
Comment (kvramyasri7) [inline packages/core/src/singleton/Storage/types.ts:11]: do we need to export this type here?

File: packages/core/src/singleton/Storage/types.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+export type StorageAccessLevel = 'guest' | 'protected' | 'private';
+
+export interface StorageConfig {
+	bucket?: string;
+	region?: string;
```
Comment (kvramyasri7) [inline packages/core/src/singleton/Storage/types.ts:8]: are we missing defaultAccessLevel here 🤔?

File: packages/core/src/singleton/Storage/types.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+type StoragePrefixResolver = (params: {
+	level: StorageAccessLevel;
```
Comment (ashwinkumar6) [inline packages/core/src/singleton/Storage/types.ts:12]: Nit: should we rename this to `accessLevel`

File: packages/core/src/singleton/Storage/types.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	identityId?: string;
```
Comment (ashwinkumar6) [inline packages/core/src/singleton/Storage/types.ts:13]: Nit: should we rename this to `targetIdentityId`

**Review by kvramyasri7 — APPROVED** [review]
LGTM

**Review by ashwinkumar6 — APPROVED** [review]

**Pull request merged**